xapi/xapi.go

```diff
@@ -152,6 +191,7 @@ func (r *partialResult) Next() (handle int64, data []types.Datum, err error) {
 
 // Close closes the sub result.
 func (r *partialResult) Close() error {
+	close(r.done)
```

Comment:
According to Go language specification, closing a closed channel causes a run-time panic:
https://golang.org/ref/spec#Close

Close can't be call multi times.
@hanfei1991